presto-main/src/main/java/com/facebook/presto/Session.java

```diff
@@ -358,11 +364,22 @@ public ClientSession toClientSession(URI server, boolean debug, Duration clientR
                 timeZoneKey.getId(),
                 locale,
                 properties.build(),
+                encodedStatements.build(),
                 transactionId.map(TransactionId::toString).orElse(null),
                 debug,
                 clientRequestTimeout);
     }
 
+    private static String utf8Encode(String string)
```

Comment:
As a separate commit, let's add functions `urlEncode` and `urlDecode` to `ResourceUtil` and update the existing usages to use them. I was thinking about that while reviewing the previous changes.